src/shapes/object.class.js

```diff
@@ -1729,12 +1729,11 @@
         canvas.backgroundColor = '#fff';
       }
       this.setPositionByOrigin(new fabric.Point(canvas.width / 2, canvas.height / 2), 'center', 'center');
-
       var originalCanvas = this.canvas;
-      canvas.add(this);
-      var canvasEl = canvas.toCanvasElement(multiplier || 1, options);
-      this.shadow = originalShadow;
+      this.set('canvas', canvas);
```

Comment:
switching from add to set('canvas') don't we miss a setCoords on the object that could be necessary for proper rendering?

Reply:
i would add a setCoords here, just to be sure we change with something is equivalent

Reply:
Also, what about we do:
canvas.objects = [this] since this is a trash-away canvas and we avoid adding an extra options to toDataUrl?

When adding this option here:
https://github.com/fabricjs/fabric.js/pull/7907/files#diff-6748525f6b3662ac6540af0ceab4cbcd5e5d44069b54ec0b1c3d9edab1e894ecR64

Is unclear that passing the objects in, those have to have a canvas reference properly set and setCoords done.
Is just an open door for misunderstandings. let's solve it inside the object.toDataUrl function and leave the rest as it is.

Reply:
Ok this is good.
Regarding setCoords - why would we need it? 
https://github.com/fabricjs/fabric.js/pull/7907/files#diff-de494ae6278212520cae6374e5a8eecaa8d08248ab53f6f30c167949186292caR1726

Reply:
i think i would needed because if the object in question has offscreen coords, it will eventually not be drawn.
Since it needs to be drawn ( the whole reason of the dataurl export ) and coords can influence a drawing skip, and since we were implicitly setting them using add, i would continue setting them.
The coords are indeed reset at the end of the function. If we don't change them, there would be at least no reason to set them again.

Reply:
> de494ae6278212520cae6374e5a8eecaa8d08248ab53f6f30c167949186292caR1726

`skipOffscreen: false,`